OpenRA.Mods.Common/Traits/ProductionParadrop.cs

```diff
@@ -0,0 +1,170 @@
+﻿#region Copyright & License Information
+/*
+ * Copyright 2007-2015 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation. For more information,
+ * see COPYING.
+ */
+#endregion
+
+using System;
+using System.Drawing;
+using System.Linq;
+using OpenRA.Activities;
+using OpenRA.Mods.Common.Activities;
+using OpenRA.Mods.Common.Traits;
+using OpenRA.Primitives;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.Common
+{
+
+	[Desc("Deliver the unit in production via paradrop.")]
+	public class ProductionParadropInfo : ProductionInfo
+	{
+		public readonly string ReadyAudio = "ReinforcementsArrived";
+		[Desc("Cargo aircraft used.")]
+		[ActorReference] public readonly string ActorType = "c17";
+
+		public override object Create(ActorInitializer init) { return new ProductionParadrop(init, this); }
+	}
+
+	class ProductionParadrop : Production
+	{
+		readonly Lazy<RallyPoint> rp;
+
+		public ProductionParadrop(ActorInitializer init, ProductionParadropInfo info)
+			: base(init, info) 
+		{
+			rp = Exts.Lazy(() => init.Self.IsDead ? null : init.Self.TraitOrDefault<RallyPoint>());
+		}
+
+		public override bool Produce(Actor self, ActorInfo producee, string factionVariant)
+		{
+			var owner = self.Owner;
+
+			// Start a fixed distance away: the width of the map.
+			// This makes the production timing independent of spawnpoint
+			var startPos = self.Location + new CVec(owner.World.Map.Bounds.Width, 0);
+			var endPos = new CPos(owner.World.Map.Bounds.Left - 5, self.Location.Y);
+
+			// Assume a single exit point for simplicity
+			var exit = self.Info.TraitInfos<ExitInfo>().First();
+
+			foreach (var tower in self.TraitsImplementing<INotifyDelivery>())
```

Comment:
This is a bad variable name because it has no real merit here (many things could be notified, it isn't always control), could you please rename it?

I'd prefer `notify`.